src/main/java/org/truffleruby/core/kernel/KernelNodes.java

```diff
@@ -1426,11 +1426,16 @@ private String getFullPath(final String featureString) {
         }
 
         private String dirname(String path) {
-            final int lastSlash = path.lastIndexOf(File.separatorChar);
-            if (lastSlash == -1) {
-                return getContext().getFeatureLoader().getWorkingDirectory();
+            if (path.charAt(0) == File.separatorChar) {
+                return path.substring(0, path.lastIndexOf(File.separatorChar));
             } else {
-                return path.substring(0, lastSlash);
+                final String workingDirectory = getContext().getFeatureLoader().getWorkingDirectory();
+                final int lastIndex = path.lastIndexOf(File.separatorChar);
+                if (lastIndex == -1) {
+                    return workingDirectory;
+                } else {
+                    return workingDirectory + "/" + path.substring(0, path.lastIndexOf(File.separatorChar));
```

Comment:
`path.lastIndexOf(File.separatorChar)` could be replaced by `lastIndex`, which would make it easier to read.

Reply:
Fixing.